### ISSUE TITLE
parallel target with vectorize requires types to be specified

### ISSUE THREAD

**Issue by gmarkall** [numba/numba#1870]
The parallel target produces normal numpy ufuncs, not numba `DUFunc` objects:

```python
@vectorize
def rel_diff_serial(x, y):
    return 2 * (x - y) / (x + y)

@vectorize(target='parallel')
def rel_diff_parallel(x, y):
    return 2 * (x - y) / (x + y)

type(rel_diff_serial)   # numba.npyufunc.dufunc.DUFunc
type(rel_diff_parallel) # numpy.ufunc
```

This means that just adding `target='parallel'` to `@vectorize` functions can break things because there won't be implementations for any types - should the parallel target produce `DUFunc`s as well?

**Comment by seibert** [comment]
Yes, we should fix this.  I think we want to be generating DUFuncs in all cases eventually.

**Comment by jzwinck** [comment]
Is this why literally nothing works when you do `numba.vectorize(target='parallel')`?  It's been more than a year--perhaps we should just add a note to the documentation saying that explicit types are mandatory with `target='parallel'`?  Right now there seems to be no mention of it at all, which leads to parallel functions being broken by default.
